gamePlay.js

```diff
@@ -0,0 +1,88 @@
+import { displayRevealedCell, revealWholeGrid, resetDisplay, getCellElem} from "./display.js";
+import Grid from "./Grid.js";
+
+
+export const changeDisplayedCell = function (celElem, cell, gridObject, game) {
+    if (game.gameOver){
+        return;
+    }
+
+    if (game.isFlaggingOn){
+        toggleFlagging(cell, celElem);
+        return;
+    }
+
+    if (cell.numMinesAround == 0){
+        gridObject.revealZeros(cell).forEach(c => displayRevealedCell(c));
+        return;
+    }
+
+    if (cell.isMine){
+        cell.isRevealed = true;
+        revealWholeGrid(gridObject);
+        game.gameOver = true;
+        return;
+    }
+
+    cell.isRevealed = true;
+    displayRevealedCell(cell);
+    return;   
+};
+
+export const clickToReveal = function (gridObject, game){
+    gridObject.cells.forEach(cell => {
+        let cellElem = getCellElem(cell);
+        cellElem.addEventListener("click", () => 
+            changeDisplayedCell(cellElem, cell, gridObject, game)
+        );
+    });
+}
+
+
+
+export const toggleFlagging = function (object, cellElem){
```

Comment:
this is a case where I think a comment is warranted and would be helpful - something explaining that `object` is either a `cell` or `something` (forgot what else it could be) - that's something that would be helpful imo